From c6af98c6bc68d682e5c4013e1112e222252d2ee2 Mon Sep 17 00:00:00 2001
From: Michel Pollet <buserror@gmail.com>
Date: Wed, 3 Mar 2010 10:09:59 +0000
Subject: [PATCH 1/1] imports: Fixes an issue with uclibc

uCLibc can be compiled with _GNU_SOURCE but without locales.
this tries to sort this mess up, it affects only this file
it seems.

Signed-off-by: Michel Pollet <buserror@gmail.com>
---
 src/mesa/main/imports.c |   14 ++++++++++++--
 1 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/src/mesa/main/imports.c b/src/mesa/main/imports.c
index 56e8195..f7d6025 100644
--- a/src/mesa/main/imports.c
+++ b/src/mesa/main/imports.c
@@ -48,7 +48,17 @@
 #include "context.h"
 #include "version.h"
 
-#ifdef _GNU_SOURCE
+#if defined(__UCLIBC__)
+ #if defined(__UCLIBC_HAS_LOCALE__)
+  #define MESA_HAS_LOCALE 1
+ #endif
+#else
+ #ifdef _GNU_SOURCE
+  #define MESA_HAS_LOCALE 1
+ #endif
+#endif
+
+#ifdef MESA_HAS_LOCALE
 #include <locale.h>
 #ifdef __APPLE__
 #include <xlocale.h>
@@ -799,7 +809,7 @@ _mesa_strdup( const char *s )
 double
 _mesa_strtod( const char *s, char **end )
 {
-#ifdef _GNU_SOURCE
+#ifdef MESA_HAS_LOCALE
    static locale_t loc = NULL;
    if (!loc) {
       loc = newlocale(LC_CTYPE_MASK, "C", NULL);
-- 
1.7.0

